src/compare-tree/key-path.js

```diff
@@ -0,0 +1,40 @@
+const KEY_DELIMETER  = '.'
+const PATH_DELIMETER = ','
+
+import {
+  castPath
+} from '../utils'
+
+function byKeyLength (a, b) {
+  return a.length > b.length ? 1 : -1
+}
+
+export function getKeyString (value) {
+  return value.join(KEY_DELIMETER)
+}
+
+export function getKeyPaths (value) {
+  if (value == null) {
+    return []
+  }
+
+  let paths = value
+
+  if (Array.isArray(value) === false) {
+    paths = `${paths}`.split(PATH_DELIMETER)
+  }
+
+  return paths.map(castPath)
```

Comment:
👍  i think i've been switching what key and path mean in my head for the last 10 minutes, should be settled now